crates/analyzer/src/symbol.rs

```diff
@@ -965,18 +973,46 @@ impl TypeKind {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
 pub struct UserDefinedType {
     pub path: Vec<StrId>,
+    pub identifier: syntax_tree::ScopedIdentifier,
```

Comment:
How about changing `path: Vec<StrId>` to `path: GenericSymbolPath` instead of adding `identifier`?
`GenericSymbolPath` probably have sufficient information, and is more structured than `ScopedIdentifier`.